### ISSUE TITLE
Finamp Android does not draw the app Edge to Edge

### ISSUE THREAD

**Issue by pohuing** [jmshrv/finamp#917]
Android can let app draw behind the navigation pill. The small indicator at the bottom of the screen if you use gesture based navigation. 

Most apps use this space either as additional screen space, drawing their app contents behind it as well, causing a scrolling list for instance to draw from the edge of your screen. Some also just put a placeholder there in a fitting color. Finamp currently does neither. Leaving the space black. Also there's rumors that the [EdgeToEdge mode will be forced by Android in Android 15.](https://www.androidauthority.com/android-15-edge-to-edge-opt-out-3467646/)

Note the black bar at the bottom:
 Album view | Player screen | Song detail bottom sheet
-|-|-
![Screenshot 2024-10-09 213348](https://github.com/user-attachments/assets/f924a962-ea90-4128-a281-f9432526496a) | ![Screenshot 2024-10-09 213401](https://github.com/user-attachments/assets/d9f77052-ed1b-4707-ae2b-fc6fcece6f45) | ![Screenshot_20241009-213334](https://github.com/user-attachments/assets/1c76d9e4-1606-47f5-9c08-4abcba78b03c)

## Possible solutions
In Flutter this can be changed with `SystemChrome` similar to how Findroid [already fixes the status bar on iOS](https://github.com/jmshrv/finamp/blob/d1845a38d1d858b0d7f1c06697e618d4750381cf/lib/main.dart#L111) by making the space transparent. This causes the `SafeArea` from `Scaffold` to be visible, creating a more coherent(imo) appearance.
```dart
await SystemChrome.setEnabledSystemUIMode(SystemUiMode.edgeToEdge);
SystemChrome.setSystemUIOverlayStyle(const SystemUiOverlayStyle(systemNavigationBarColor: Colors.transparent));
```

I am not entirely sure if these system chrome changes stick around during the entire app lifetime. So there might be a need for supplying a `WidgetsBindingObserver`. I use these solutions in conjunction in my private projects without issues(though I only have an Android phone):

```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

/// Class to set the system UI overlay style to edge to edge.
class UIOverlaySetterObserver extends WidgetsBindingObserver {
  @override
  void didChangeAppLifecycleState(AppLifecycleState state) async {
    switch (state) {
      case AppLifecycleState.resumed:
        SystemChrome.setSystemUIOverlayStyle(
          const SystemUiOverlayStyle(
            systemNavigationBarColor: Colors.transparent,
          ),
        );
        await SystemChrome.setEnabledSystemUIMode(SystemUiMode.edgeToEdge);
        break;
      default:
        break;
    }
  }
}
```
which would be used in main.dart
```dart
final binding = WidgetsFlutterBinding.ensureInitialized();
binding.addObserver(UIOverlaySetterObserver());
```

**Comment by pohuing** [comment]
I would like to make PR but right now my Flutter installation is FUBAR and my internet too slow to reinstall the entire toolchain. So that would have to wait a bit

**Comment by pohuing** [comment]
flutter 3.24.4 magically fixed my gradle woes and I finally got around to trying this out.

**Comment by pohuing** [comment]
Something broke this in a recent update. Opening the player screen resets the overlay mode

**Comment by pohuing** [comment]
Funnily enough my observer works, locking the screen and unlocking returns you to edge to edge :D

**Comment by pohuing** [comment]
Reviewing the changes since this was merged I can't make out an obvious reason for the regression. I'll take a closer look in a week

**Comment by Chaphasilor** [comment]
@pohuing this was done to fix contrast issues on the player screen. We probably need to manually revert back to fullscreen mode when leaving the player screen...

**Comment by Chaphasilor** [comment]
You can find the related issue here: https://github.com/jmshrv/finamp/issues/798

**Comment by pohuing** [comment]
There might be a better solution of only enabling the top status bar.

**Comment by Ambient-Impact** [comment]
This looks great, but I'm wondering if it would be possible to add a bit of a shadow (or blur?) behind the navigation bar at the bottom and similarly for the notification bar at the top? I find it feels kind of cluttered or busy visually when scrolling through the library, playlists, etc. Could even be optional in the settings, since I can almost hear future comments from people who disagree with such a change. 😂

**Comment by Chaphasilor** [comment]
We'll keep it the way it is for now. This full screen setup seems to have been the cause of a hang on startup that has been happening on several beta versions, so I don't want to mess with this any more than needed 😅

**Comment by Ambient-Impact** [comment]
Fair enough. I know how much of a nightmare keeping so many proverbial plates up in the air can be!
